gnovm/stdlibs/std/native.go

```diff
@@ -16,7 +16,13 @@ func AssertOriginCall(m *gno.Machine) {
 }
 
 func IsOriginCall(m *gno.Machine) bool {
-	return len(m.Frames) == 2
+	n := m.NumFrames()
+	if n == 0 {
+		return false
+	}
+	firstPkg := m.Frames[0].LastPackage
+	isMsgCall := firstPkg != nil && firstPkg.PkgPath == "main"
```

Comment:
Does it make sense to just type check against the existing MsgCall type, if it is available in this context?

Reply:
Good point, I wasn't very happy with this assertion, because it's a little weak.
I've checked the context and we do have access to the `Msg` interface, so we can replace this line by:
```
isMsgCall := m.Context.(ExecContext).Msg.Type() == "exec"
```
Do you think also it's better ? I ask because it requires to update the unit tests (the txtar tests are running well with this change).

Reply:
> In what case would firstPkg be nil?

For the first frame, probably never. That being said, a frame can have a `LastPackage == nil` when it represents a block inside a function, like a for loop (I'm not 100% sure about that).